Update CalculationBaseType.Enum.al

AydinKelky opened the pull request:
#### Summary 
Update CalculationBaseType.Enum.al to Extensible=True
#### Fixes #4092 









Fixes [AB#596253](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/596253)

NKarolak commented:
Making an Enum extensible typically (but not always) requires adding events to handle possible custom values. Have you already checked where and how this Enum is used?

AydinKelky commented:
@NKarolak Thanks for the clarification. There are some parts that use this enum as a parameter(case or if). Should I create events for the parts that might be causing a problem?

AydinKelky commented:
@NKarolak It would be great if someone from Microsoft could help me with this. Thank you for your help as well :)

JesperSchulz commented:
> @JesperSchulz Could you please help me with this issue. It's been a long time since I created this Pull and we really need to expand this enum because for migration from Singhammer to Microsoft, many of our programming depends on this enum extension. I would really appreciate it if you help us make this issue resolved. Thank you in advance

Sorry, I've been on vacation the past month, but now I'm back and I'll try to get some traction on those PRs.

JesperSchulz commented:
CI is failing. Could you try pulling from main to get your branch updated?

AydinKelky commented:
@JesperSchulz I updated my Branch.

JesperSchulz commented:
> @JesperSchulz I updated my Branch.

Thanks 🙏 Let me try to run our pipeline once more!

AydinKelky commented:
@JesperSchulz I updated my branch but I checked it again now and it says it is out-of-date. What I did was I went to AydinKelky/BCApps and updated Sync fork. Please let me know if I did the update correctly. if not, please tell me what to do
